### PR TITLE
Update table of contents index and add style guide to docs

### PR DESCRIPTION
**Pull request by willingc**
This PR fixes some minor issues noted by @paulproteus.

The draft style guide based on Kenneth Reitz' Python Guide project.

The README.rst was removed from the base docs directory since RTD documentation indicated that it may override the Table of Contents in index.rst. CONTRIBUTIONS.rst has also been removed now that there is a Contributor's Guide in the docs.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-Contributing to OpenHatch
-=========================
-
-Before opening a pull request or submitting a patch, please `read the guidelines for contributing`_.
-
-If you submit more than one pull request, you'll find that it's helpful to use a topic branch to avoid duplicating commits.
-
-.. _read the guidelines for contributing: http://openhatch.readthedocs.org/en/latest/getting_started/getting_started.html
-
-.. Further changes to this file should be made using the limited subset of rST syntax that GitHub supports.
```
Comment (paulproteus) [inline CONTRIBUTING.rst]: For what it's worth, GitHub itself uses this -- see also https://github.com/blog/1184-contributing-guidelines

File: docs/CONTRIBUTING.rst [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-Contributing to OpenHatch
-=========================
-
-Before opening a pull request or submitting a patch, please `read the guidelines for contributing`_.
-
-If you submit more than one pull request, you'll find that it's helpful to use a topic branch to avoid duplicating commits.
-
-.. _read the guidelines for contributing: http://openhatch.readthedocs.org/en/latest/getting_started/getting_started.html
-
-.. Further changes to this file should be made using the limited subset of rST syntax that GitHub supports.
```
Comment (paulproteus) [inline docs/CONTRIBUTING.rst]: +1 on this deletion -- I think the one at the root makes sense, but the one in docs/ doesn't.

File: docs/README.rst [hunk 3]
```diff
@@ -1,16 +0,0 @@
...
-
-.. image:: https://coveralls.io/repos/openhatch/oh-mainline/badge.png?branch=master
-  :target: https://coveralls.io/r/openhatch/oh-mainline?branch=master
-
-
-Thanks for checking out our repository. We love this project; hope you do too.
-
-You can find our documentation in the docs/ directory. It's automatically rendered online at http://openhatch.readthedocs.org/en/latest/index.html
-
-Is readthedocs not working?  Let us know by emailing hello@openhatch.org or finding us on IRC at #openhatch on irc.freenode.net.
```
Comment (paulproteus) [inline docs/README.rst]: Is there still a README.rst at the top level? If so, +1 on this deletion.

File: docs/advanced/styleguide.rst [hunk 4]
```diff
@@ -0,0 +1,192 @@
+.. _guide-style-guide:
+
+=============================
+The Guide Style Guide - DRAFT
+=============================
+
+.. note:: This document is attributed to Kenneth Reitz's excellent Python
```
Comment (paulproteus) [inline docs/advanced/styleguide.rst]: I'm -0 on including this by content. Can we get away with linking to Kenneth Reitz's rST style guide, and then (optionally) stating any differences?

Also, we should clarify somewhere if this is a style guide for the docs, or a style guide for the Python code.

File: docs/conf.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
-version = '01.01.01'
+version = '01.01.02'
```
Comment (paulproteus) [inline docs/conf.py]: +0, okay. I wouldn't bother bumping this; I don't think we rely on it anywhere.

We used to use this `version` flag when we were developing on a monthly/six-monthly/etc. cadence, and back then, we used to also write release notes:

See e.g.

http://lists.openhatch.org/pipermail/devel/2011-April/001838.html

http://blog.openhatch.org/2011/openhatch-0-11-03-is-released/

File: docs/getting_started/index.rst [hunk 6]
```diff
@@ -1,6 +1,6 @@
-==================
-Contributors Guide
-==================
+===================
+Contributor's Guide
```
Comment (paulproteus) [inline docs/getting_started/index.rst]: +1

File: docs/index.rst [hunk 7]
```diff
@@ -1,9 +1,11 @@
-====================
-Welcome to OpenHatch
-====================
+====================================
+Welcome to OpenHatch's Documentation
```
Comment (paulproteus) [inline docs/index.rst]: +1

File: docs/index.rst [hunk 8]
```diff
@@ -1,9 +1,11 @@
...
+====================================
 
 `OpenHatch <http://openhatch.org/>`_ values open communication. We believe
-documentation is an important part of a welcoming community.
+documentation is an important part of a welcoming community. If you would
+like to see some of the projects that we are currently working on, please
+visit the OpenHatch website <http://openhatch.org>.
```
Comment (paulproteus) [inline docs/index.rst]: Great clarifying text!

File: CONTRIBUTING.rst (hunk 1 above)
Comment (paulproteus) [inline CONTRIBUTING.rst]: Arguably we should document the purpose of those files somewhere, to avoid this confusion in the future!

**Comment by paulproteus** [comment]
I'm going to merge this as-is, even though I have minor concerns about the removal of CONTRIBUTING.rst.

Life is short and I haven't merged enough stuff lately. Huge thanks for working on this @willingc .

File: CONTRIBUTING.rst (hunk 1 above)
Comment (willingc) [inline CONTRIBUTING.rst]: Yes we should. I think that I can add back CONTRIBUTING.rst in the main directory.

File: docs/README.rst (hunk 3 above)
Comment (willingc) [inline docs/README.rst]: Yup left the top level

File: docs/advanced/styleguide.rst (hunk 4 above)
Comment (willingc) [inline docs/advanced/styleguide.rst]: I like your suggestion. I will revise tomorrow :)

**Pull request merged**